ddl/ddl_api.go

```diff
@@ -3258,6 +3266,20 @@ func checkIsDroppableColumn(ctx sessionctx.Context, t table.Table, spec *ast.Alt
 	return true, nil
 }
 
+func checkDropVisibleColumnCnt(t table.Table, columnCnt int) error {
+	tblInfo := t.Meta()
+	visibleColumCnt := 0
+	for _, column := range tblInfo.Columns {
+		if !column.Hidden {
+			visibleColumCnt++
+		}
+	}
+	if visibleColumCnt == columnCnt {
+		return ErrTableMustHaveColumns
+	}
+	return nil
```

Comment:
how about 
```go
for _, column := range tblInfo.Columns {
	if !column.Hidden {
		visibleColumCnt++
	}
    if visibleColumCnt > columnCnt  {
        reutrn nil
    }
}
return ErrTableMustHaveColumns
```

Reply:
good idea, This can reduce the number of cycles
done